ndcube/ndcube.py

```diff
@@ -525,7 +526,50 @@ def __repr__(self):
 
 
 class NDCube(NDCubeBase, NDCubePlotMixin, astropy.nddata.NDArithmeticMixin):
-    pass
+
+    def explode_along_axis(self, axis):
+        """
+        Separates slices of NDCubes in sequence along a given cube axis into a NDCubeSequence
+        of (N-1)DCubes.
+
+        Parameters
+        ----------
+        axis : `int`
+            The axis along which the data is to be changed.
+
+        Returns
+        -------
+        result : `ndcube_sequence.NDCubeSequence`
+
+        """
+        # If axis is -ve then calculate the axis from the length of the dimensions of one cube
+        if axis < 0:
+            axis = len(self.dimensions) + axis
+        # the range of the axis that needs to be sliced
+        range_of_axis = self.data.shape[axis]
+        # To store the resultant cube
+        result_cubes = []
+        # We need to convert extra_coords dictionnary to a liste of tuple to create a new NDCube
+        extra_coords_formated = utils.cube.convert_extra_coords_dict_to_input_format(
+            self.extra_coords, missing_axis=[False, False, True])
+        # Creating a (N-1)Dcube list
+        for i in range(range_of_axis):
+            cube = self._new_instance(
+                self[i].data, self[i].wcs, uncertainty=self[i].uncertainty, unit=self.unit,
+                meta=self.meta, mask=self[i].mask, missing_axis=[False, False, True],
+                extra_coords=extra_coords_formated)
+            result_cubes.append(cube)
+        return NDCubeSequence(result_cubes, meta=self.meta)
```

Comment:
I think this method could be rewritten with the following:
```python
result_cubes = []
cube_slices = [slice(None, None, None)] * len(self.data.ndim)
for i in range(self.data.shape[axis]):
    cube_slices[axis] = i
    result_cubes.append(self[cube_slices])
return NDCubeSequence(result_cubes, common_axis=axis)
```

Reply:
You make me understand these lines just now ! Of course, this is the best way.